### DISCUSSION TITLE
想请教一下如何让NoneBot的FastAPI的监听端口不同于gocqhttp的webUI的端口

### DISCUSSION THREAD

**Discussion by sanmusen214** [nonebot/discussions#308] [问答] [answered]
按照https://v2.nonebot.dev/docs/tutorial/add-custom-api 为例使用的FastAPI
使用发现FastAPI的端口和nonebot-plugin-gocqhttp的webUI的端口一样，导致在公网可以通过FastAPI各个接口的端口后加上/go-cqhttp/直接通过webUI操作已登录的机器人，请问有什么办法能让这两个功能对应两个端口

**Comment by yanyongyu** [comment, marked as answer]
不可以，你可以通过路由转发控制，或者给机器人添加token

**Reply by sanmusen214** [reply]
感谢回答，不知后续是否有在项目配置里独立设置Driver的端口的计划？

**Reply by yanyongyu** [reply]
这是不可能实现的，无法监听两个端口

**Reply by mnixry** [reply]
可以参考 https://github.com/mnixry/nonebot-plugin-gocqhttp/issues/42#issuecomment-1042504781 配置一个反代来进行公网开放
